core/src/main/scala/org/apache/spark/util/JsonProtocol.scala

```diff
@@ -394,10 +394,13 @@ private[spark] object JsonProtocol {
 
   /** Convert executor metrics to JSON. */
   def executorMetricsToJson(executorMetrics: ExecutorMetrics): JValue = {
-    val metrics = ExecutorMetricType.values.map{ metricType =>
-      JField(metricType.name, executorMetrics.getMetricValue(metricType))
-     }
-    JObject(metrics: _*)
+    val metrics = for {
+      (m, _) <- ExecutorMetricType.definedMetricsAndOffset
+    } yield {
```

Comment:
there's nothing wrong with this, but I think spark is biased towards using `foreach`, `map`, & `flatMap` over scala's `for / yield` unless there is a particular reason to use it.